### DISCUSSION TITLE
멀티플레이 걷기

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#620] [5. Q&A]
미리 말씀드립니다. 
테스트 환경은 제페토에서 제공해 주신 zepeto-multiplay-example-main 이고, 별도의 스크립트 변경은 없습니다.
멀티플레이 동작으로 상대방 폰에서 화면을 캡쳐한 것이고, 
PC 는 맥입니다.
제페토 멀티플레이에 어느정도 익숙합니다. 

그간 별 문제 없는 것으로 생각되어 지나쳤는데, 어느순간 거슬리는 동작이 있어 여쭤봅니다.
멀티플레이시 내가 걷기를 하면 상대방의 폰에서도 걷기를 해야 정상인것 같은데.
아래 화면 처럼 이상한 동작을 합니다. 

스크립트의 
OnUpdatePlayer(sessionId: string, player: Player) 함수를 살펴보면

상대방 플레이어의 동작이 아래 함수로 작성되어 있습니다. 
zepetoPlayer.character.MoveToPosition(position); 

별도의 걷기 동작이 없는 것으로 봐서 내부에 있지않나 생각하는데요.
뛸때는 아무 문제 없었습니다. 

문제는 걷기 동작에서 상태를 살펴보면 
zepetoPlayer.character.CurrentState 가 1과 3을 반복합니다.  
1은 idle 이고, 3은 Run 이기 때문에 
화면에서와 같이 가다 서다를 반복하는것 같습니다.

이 문제는 제가 해결해야 하는건가요? 아님 제페토에서 해결해줘야 하는 건가요?

설정이 잘못되었다고 생각하실까봐 다시 한번 말씀드리면 
테스트 환경은 제페토에서 제공해 주신 zepeto-multiplay-example-main 이고, 별도의 스크립트 변경은 없습니다.
같은 프로젝트로 테스트해 보시면 바로 알 수 있을것 같습니다.

https://user-images.githubusercontent.com/30133892/163401702-048f2a70-d1ae-4cfb-869f-13f86c288bbf.mov

**Comment by x4cf** [comment]
샘플은 샘플일 뿐이라서 걷기처리를 따로 만들어두지 않은것으로 알고있습니다. 정상적으로 작동되는 장면으로 보입니다 :D

**Comment by Joyounwoo** [comment]
결론부터 말씀드린다면 직접 로직을 수정하셔야 합니다.
제페토 멀티플레이 샘플에서는 서버로부터 받은 플레이어 좌표를 기반으로 다른 유저의 제페토 캐릭터에게 MoveToPosition 함수를 사용하여 이동시킵니다.
이 때, MoveToPosition 함수는 무조건 Run 애니메이션을 이용해서 이동시킵니다. 그렇기에 캐릭터의 애니메이션이 Run 상태입니다.

이 부분을 수정하시면, 캐릭터가 정상적으로 걷는 모습으로 보일 수 있겠습니다만 그래도 여전히 캐릭터가 멈칫하실 겁니다.
MoveToPosition 함수를 사용하면 캐릭터는 무조건 달리는 속도를 기준으로 이동하게 됩니다. 자신의 클라이언트에서는 느린 속도로 걸어 이동하지만,
타 유저의 클라이언트에선 자신의 캐릭터가 전달된 좌표에 더 빠르게 도달하게 됩니다. 그러면 다음 좌표가 전달되기 전에 잠깐 멈춰있게 됩니다.
이게 좌표가 전달되는 텀 마다 발생하기에 저렇게 멈칫거리게 된다고 이해하시면 되겠습니다.

**Reply by smilejsu82** [reply]
@Joyounwoo 답변 감사합니다 ~ 👍
